front/lib/api/data_sources.ts

```diff
@@ -729,14 +729,14 @@ export async function createDataSourceWithoutProvider(
     space,
     name,
     description,
-    conversation,
+    conversationId,
   }: {
     plan: PlanType;
     owner: WorkspaceType;
     space: SpaceResource;
     name: string;
     description: string | null;
-    conversation?: ConversationType;
```

Comment:
What's the motivation here? (Unclear from the rest of the diff and we really really want to avoid passing ModelIds around when possible, but I know you know :-))

Reply:
Where are you saving that query to be clear?

Reply:
Won't fix my dep cycle :(

Let's look at this in another PR to avoid blocking this one.

Reply:
Let's create a task. ModelId in interfaces should really be fought like cholera

Reply:
> Let's create a task.

Sure.

> ModelId in interfaces should really be fought like cholera

Why ?

Reply:
Creates a dependency in our code to the storage technology we use. Goal is for that to be entirely mediated by resources.

Reply:
I understand (and agree with) the goal, but:

- `ModelId` is just an alias for number.
- We are talking about passing "XXXType" which is not a "Ressource" either, and for fetching the ressource, we need an id as well (or a sid but in term of "dependency to the storage technology" it's just obfuscation).

So I'm not super convinced by the solution of using a XXType : i don't mind doing it but I don't think it solve the problem and it comes with some caveat (such as, fetching things we don't need or creating custom "light" version).